File: snuggle_server/src/main/java/com/ssafy/snuggle/model/service/UserService.java
package com.ssafy.snuggle.model.service;

import com.ssafy.snuggle.model.dto.User;

public interface UserService {

	/**
	 * 사용자 정보를 DB에 저장하고 저장된 건수를 리턴한다.
	 * 
	 * @param user
	 */
	public int join(User user);

	/**
	 * id, pass에 해당하는 User 정보를 반환한다.
	 * 
	 * @param user_id
	 * @param password
	 * @return 조회된 User 정보를 반환한다.
	 */
	public User login(String user_id, String password);

	/**
	 * 해당 아이디가 이미 사용 중인지를 반환한다.
	 * 
	 * @param user_id
	 * @return
	 */
	public boolean isUsedId(String user_id);

	/**
	 * id 에 해당하는 User 정보를 반환한다.
	 * 
	 * @param user_id
	 * @return 조회된 User 정보를 반환한다.
	 */
	public User selectUser(String user_id);

}

File: snuggle_final_app/app/src/main/java/com/ssafy/snuggle_final_app/data/service/UserService.kt
package com.ssafy.snuggle_final_app.data.service

import com.ssafy.snuggle_final_app.data.model.dto.User
import com.ssafy.snuggle_final_app.data.model.response.UserResponse
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query
import java.util.Locale.Category

interface UserService {

    @POST("snuggle/user")
    suspend fun insert(@Body body: User): Boolean

    @GET("snuggle/user/info")
//    suspend fun getUserInfo(@Query("userId") id: String): Response<UserResponse>
    suspend fun getUserInfo(@Query("id", encoded = true) id: String): Response<UserResponse>

    @GET("snuggle/user/isUsed")
    suspend fun isUsedId(@Query("userId") id: String): Boolean

    @POST("snuggle/user/login")
    suspend fun login(@Body body: User): User

}
